src/core/varlink-common.c

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "json-util.h"
+#include "rlimit-util.h"
+#include "varlink-common.h"
+
+int rlimit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        const struct rlimit *rl = userdata;
+        struct rlimit buf = {};
+        int r;
+
+        assert(ret);
+        assert(name);
+
+        if (!rl) {
+                int z = rlimit_from_string(name);
+                assert(z >= 0 && z < _RLIMIT_MAX);
```

Comment:
hmm, this funciton is somewhat generic, and I think it would be better to treat a bad `name` being passed in as a runtime error rather than a programming error, hence please replace this assert by `if (z < 0) return z;`